packages/blockchain/src/blockchain.ts

```diff
@@ -1172,8 +1176,13 @@ export class Blockchain implements BlockchainInterface {
     return this.dbManager.getHeader(hash, number)
   }
 
-  protected checkAndTransitionHardForkByNumber(number: bigint, td?: BigIntLike): void {
+  protected checkAndTransitionHardForkByNumber(
+    number: bigint,
+    blockHeader: BlockHeader,
+    td?: BigIntLike
+  ): void {
     this._common.setHardforkByBlockNumber(number, td)
+    blockHeader._common.setHardforkByBlockNumber(number, td)
```

Comment:
Because `common` is copied, now we have to also explicitly set hardfork in the blockHeader. There is a high chance we have more bugs which are the same (common hardfork changes in one package, where it should also change in another package, but because it is copied this is not done and we should thus explicitly change the "chain" of all common hardforks of relevant package objects).